pkg/ovs/openflow/ofctrl_group.go

```diff
@@ -85,6 +85,50 @@ func (g *ofGroup) GetBundleMessage(entryOper OFOperation) (ofctrl.OpenFlowModMes
 	return message, nil
 }
 
+func (g *ofGroup) GetBundleMessages(entryOper OFOperation, maxBucketsPerMessage int) ([]ofctrl.OpenFlowModMessage, error) {
+	var operation int
+	switch entryOper {
+	case AddMessage:
+		operation = openflow15.OFPGC_ADD
+	case ModifyMessage:
+		operation = openflow15.OFPGC_MODIFY
+	case DeleteMessage:
+		operation = openflow15.OFPGC_DELETE
+	}
+
+	// If the operation is to delete the group or empty the group, create the bundle message directly.
+	if entryOper == DeleteMessage || len(g.ofctrl.Buckets) == 0 {
+		return []ofctrl.OpenFlowModMessage{g.ofctrl.GetBundleMessage(operation)}, nil
```

Comment:
Why delete the group if buckets number is 0 even if `entryOper!=DeleteMessage`?

Reply:
When `entryOper!=DeleteMessage`, it is used to empty a group, rather than delete the group.